Mathlib/Order/Incomparable.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Incomparability relation
+
+Two values in a preorder are incomparable whenever `¬ a ≤ b` and `¬ b ≤ a`.
+
+## Main declarations
+
+* `IncompRel`: The incomparability relation. `IncompRel r a b` means that `a` and `b` are related in
+  neither direction by `r`.
+-/
+
+open Function OrderDual
+
+variable {α : Type*} {a b c d : α}
+
+section Relation
+
+variable (r : α → α → Prop)
+
+/-- The incomparability relation `IncompRel r a b` means `¬ r a b` and `¬ r b a`. -/
+def IncompRel (a b : α) : Prop :=
+  ¬ r a b ∧ ¬ r b a
+
+theorem IncompRel.not_le [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ a ≤ b := h.1
+theorem IncompRel.not_ge [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ b ≤ a := h.2
+theorem LE.le.not_incompRel [LE α] (h : a ≤ b) : ¬ IncompRel (· ≤ ·) a b := fun h' ↦ h'.not_le h
```

Comment:
Can this move out of the `Relation` section? It's supposed to be about an arbitrary relation, not about `LE`

Reply:
I'll put them at the very end, like I did with `AntisymmRel`.